### PR TITLE
KeyUsage requirements for the delegation certificate

### PR DESCRIPTION
**Pull request by cjpatton**
Based on #5 

One of the design criteria for this standard is to isolate the TLS with the DC extension from other protocols. In this PR, I propose explicitly restricting the X.509 KeyUsage of the end-entity certificate.

This change is made in the last commit: https://github.com/tlswg/tls-subcerts/commit/9bb99f847c41e66420878e551c2df2170a36d29c. The two commits that proceed it don't actually change the semantics of the standard, but clean up the language a bit. If the KeyUsage change isn't adopted, we might consider merging the other two commits.

### CODE REVIEW

File: draft-ietf-tls-subcerts.md [hunk 1]
```diff
@@ -333,19 +333,24 @@ complexity to the TLS stack).
...
 
 Conforming CAs MUST mark this extension as non-critical. This allows the
 certificate to be used by service owners for clients that do not support
 certificate delegation as well and not need to obtain two certificates.
 
+The client MUST NOT accept a delegated credential unless the server's end-entity
+certificate satisfies the following criteria:
+
+* It has the DelegationUsage extension.
+* It has the digitalSignature key usage enabled (see the Keyusage type in
```
Comment (siyengar) [inline draft-ietf-tls-subcerts.md]: i don't this this is necessary any more right? since we require an explicit extension.

Comment (cjpatton) [reply draft-ietf-tls-subcerts.md]: The extension isn't critical, so a delegation certificate could, in theory, be used as a decryption oracle.

Comment (cjpatton) [reply draft-ietf-tls-subcerts.md]: Of course, this might not matter much in practice; but it seems like the "right thing to do" in theory.

**Comment by siyengar** [comment]
if it's not too painful, could you move this particular diff above all the others, I think I want to have some more discussion on this one. If its painful, we can just resolve this one first

**Comment by cjpatton** [comment]
Done, @siyengar.

**Pull request merged**